applications/luci-app-p910nd/htdocs/luci-static/resources/view/p910nd.js

```diff
@@ -0,0 +1,278 @@
+				return;
+
+			if (!L.isObject(devices[dev].flags) || !Array.isArray(ip6addrs) || devices[dev].flags.loopback)
+				return;
+
+			//list IPv4
+			for (var i = 0; i < ip4addrs.length; i++) {
+				if (!L.isObject(ip4addrs[i]) || !ip4addrs[i].address)
+					continue;
+
+				o.value(ip4addrs[i].address, E([], [
+					ip4addrs[i].address, ' (', E('strong', {}, [dev]), ')'
+				]));
+			}
+			//list IPv6
+			for (var i = 0; i < ip6addrs.length; i++) {
+				if (!L.isObject(ip6addrs[i]) || !ip6addrs[i].address)
+					continue;
+
+				o.value(ip6addrs[i].address, E([], [
+					ip6addrs[i].address, ' (', E('strong', {}, [dev]), ')'
+				]));
+			}
+
+		});
+
+		o = s.option(form.Flag, 'bidirection', _('Bi-directional'), _('Whether this print port is bi-directional.'));
+
+		o = s.option(form.Flag, 'runas_root', _('Root'), _('Run as root: overrides running as user p910nd, group lp.'));
+
+		o = s.option(form.Flag, 'mdns', _('mDNS'), _('Whether to advertise this printer via mDNS/Bonjour/ZeroConf. Note: only advertises one printer on this host via mDNS'));
+
+		/* See https://developer.apple.com/bonjour/printing-specification/bonjourprinting-1.2.1.pdf for
+		 * a description of the Bonjour fields a la Apple.
+		 */
+		o = s.option(form.Value, 'mdns_ty', _('Description'), _('User readable description of maker and model.'));
+		o.placeholder = 'HP Color LaserJet CP2025dn';
+		o.rmempty = true;
+		o.optional = true;
+
+		o = s.option(form.Value, 'mdns_note', _('Location'));
+		o.placeholder = 'Wardrobe';
+		o.rmempty = true;
+		o.optional = true;
+		o.modalonly = true;
+
+		o = s.option(form.Value, 'mdns_product', _('Product'), _('the value of the “Product” stored in the driver’s PPD file (including the parentheses). E.g. (MFG MODEL).'));
+		o.placeholder = '(Color LaserJet CP2025dn)';
+		o.rmempty = true;
+		o.optional = true;
+
+		o = s.option(form.Value, 'mdns_mfg', _('Manufacturer'), _('the value of the MANUFACTURER (MFG) from the printer’s IEEE1284 device ID.'));
+		o.placeholder = 'Manufacturer';
+		o.value('Brother');
+		o.value('Canon');
+		o.value('Epson');
+		o.value('Hewlett-Packard');
+		o.value('Hitachi');
+		o.value('Kyocera');
+		o.value('Lexmark');
+		o.value('OKI');
+		o.value('Ricoh');
+		o.value('Samsung');
+		o.value('Xerox');
+		o.rmempty = true;
+		o.optional = true;
+
+		o = s.option(form.Value, 'mdns_mdl', _('Model'), _('the value of the MODEL (MDL) from the printer’s IEEE1284 device ID.'));
+		o.placeholder = 'CP2025dn';
+		o.rmempty = true;
+		o.optional = true;
+
+		o = s.option(form.Value, 'mdns_cmd', _('Command'), _('the value of the COMMAND SET (CMD) from the printer’s IEEE1284 device ID.'));
+		o.placeholder = 'PS,PDF17,PCL';
+		o.rmempty = true;
+		o.optional = true;
+		o.modalonly = true;
+
+		/* Some print devices need an extra helping hand. e.g. load a firmware before it will run. Do that
+		 * in a hotplug.d/usb script
+		 */
+		s.handleHotplugScripts = function() {
```

Comment:
Nice way to get in as root :)

Agreed - managing hotplug.d somehow would be a nice addition. I tacked the hotplug scripting bit on because my printer requires a firmware load before it does anything useful... I thought to abstract that all away, but, time....